src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -102,6 +102,7 @@ private static LspDiagnostic[] FilterHTMLDiagnostics(
 
         var filteredDiagnostics = unmappedDiagnostics
             .Where(d =>
+                !InRazorComment(d, sourceText, syntaxTree) &&
```

Comment:
>InRazorComment [](http://example.com/codeflow?start=17&length=14)

Legacy Razor editor used to filter out any diagnostics in style blocks if line contained any Razor span. In relied on projection spans to do so.

I wonder if we can make this more generic and filter out any html diagnostics if it comes from a line that contains any Razor span and is inside of style block or script block. Current LSP HTML editor code can't know where Razor spans are since it just sees plain text.

Reply:
We have various filtering for style blocks etc. when the diagnostics original from a Razor syntax that is likely to be misunderstood:
https://github.com/davidwengier/razor/blob/NoDiagnosticsInComments/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Diagnostics/RazorTranslateDiagnosticsService.cs#L242-L249

Razor comments just seemed like something worth going very broad with.